client_plugin/drivers/vfile/kvstore/etcdops/etcdops.go

```diff
@@ -289,20 +305,100 @@ func (e *EtcdKVS) joinEtcdCluster(leaderAddr string) error {
 	}
 
 	// start the routine for joining an etcd cluster
-	go etcdService(lines)
+	e.etcdStartService(lines)
 
 	// check if successfully joined the etcd cluster, then start the watcher
 	return e.checkLocalEtcd()
 }
 
-// etcdService function starts a routine of etcd
-func etcdService(cmd []string) {
-	_, err := exec.Command("/bin/etcd", cmd...).Output()
+// leaveEtcdCluster function is called when a manager is demoted
+func (e *EtcdKVS) leaveEtcdCluster() error {
+	nodeAddr := e.nodeAddr
+	nodeID := e.nodeID
+
+	etcd, err := addrToEtcdClient(nodeAddr)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"nodeAddr": nodeAddr,
+				"nodeID": nodeID},
+		).Error("Failed to create ETCD client from own address")
+		return err
+	}
+	defer etcd.Close()
+
+	// list all current ETCD members
+	ctx, cancel := context.WithTimeout(context.Background(), etcdRequestTimeout)
+	lresp, err := etcd.MemberList(ctx)
+	cancel()
+	if err != nil {
+		log.WithFields(
+			log.Fields{"nodeAddr": nodeAddr,
+				"error": err},
+		).Error("Failed to list member for ETCD")
+		return err
+	}
+
```

Comment:
Please add a comment about what is "peerAddr", and it could be helpful with an example